lib/src/firebase_auth_mocks_base.dart

```diff
@@ -13,9 +14,14 @@ class MockFirebaseAuth implements FirebaseAuth {
   late Stream<User?> userChangedStream;
   MockUser? _mockUser;
   User? _currentUser;
-
-  MockFirebaseAuth({bool signedIn = false, MockUser? mockUser})
-      : _mockUser = mockUser {
+  final AuthExceptions? _authExceptions;
```

Comment:
I like your choice of making this and AuthException's fields final. It makes the code simple. That means it will keep throwing the same exceptions. That should be enough for most users.